db/builder.cc

```diff
@@ -136,7 +137,8 @@ Status BuildTable(
       bool use_direct_writes = file_options.use_direct_writes;
       TEST_SYNC_POINT_CALLBACK("BuildTable:create_file", &use_direct_writes);
 #endif  // !NDEBUG
-      IOStatus io_s = NewWritableFile(fs, fname, &file, file_options);
+      FileOptions cur_file_opts(file_options, ChecksumType::kCRC32c);
```

Comment:
Just to clarify my previous comment, is there any reason why ```ChecksumType::kCRC32c``` needs to be explicitly specified? Can it be the default in ```FileOptions```? Alternatively, it could be set in ```file_options_``` and ```file_options_for_compaction_``` in the ```DBImpl``` constructor, rather than everywhere ```NewWritableFile``` is called..

Reply:
It might be better to set ChecksumType::kCRC32c as the default in FileOptions. Most likely, RocksDB internal will only use it for the handoff. If other types are supported in the future, we can set it explicitly. Thanks for the suggestion.